core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -6369,6 +6369,13 @@ protected function updateRoot(array $root)
 			}
 		}
 
+		// $this->root might not have a correct order here, let's make sure it's ordered by sorting but only in case there are no visible root trails (aka
+		// the array does contain only top-level ids)
+		if (empty($this->visibleRootTrails) && $this->Database->fieldExists('sorting', $table))
```

Comment:
```suggestion
		if ($this->root && empty($this->visibleRootTrails) && $this->Database->fieldExists('sorting', $table))
```
We need to skip for empty arrays here I think, otherwise we’d get a syntax error.

Reply:
as far as I know, an empty `root` array is never valid. I had to work around this using `[0]` in a lot of extensions 🙃

Reply:
I think it's a smart thing to do anyway though :)

Reply:
> as far as I know, an empty `root` array is never valid.

https://github.com/contao/contao/blob/d7d396df7c7a758b957fe823663c6929f136cb8e/core-bundle/src/Resources/contao/drivers/DC_Table.php#L3745